src/In.ProjectEKA.HipService/DataFlow/DataEntryFactory.cs

```diff
@@ -47,25 +49,31 @@ public DataEntryFactory()
             var randomKey = EncryptorHelper.GenerateRandomKey();
 
             var processedEntries = new List<Entry>();
-            foreach (var bundle in entries.Bundles)
```

Comment:
Can't we loop over bundle only as bundle itself has the CC I believe. What made you think this is the way we should iterate?

Reply:
But here, Bundles is the Dictionary of CareContextReference to the List of the Bundles

Reply:
Because of this if we choose to loop over bundle then the code will be Little Messy I assume